src/main/java/com/google/finapp/FinAppService.java

```diff
@@ -102,6 +103,30 @@ public void createCustomerRole(
     responseObserver.onCompleted();
   }
 
+  @Override
+  public void moveAccountBalance(
+      MoveAccountBalanceRequest request,
+      StreamObserver<MoveAccountBalanceResponse> responseObserver) {
+    Map<ByteArray, BigDecimal> accountBalances;
+    ByteArray fromAccountId = ByteArray.copyFrom(request.getFromAccountId().toByteArray());
+    ByteArray toAccountId = ByteArray.copyFrom(request.getToAccountId().toByteArray());
+    try {
+      accountBalances =
+          spannerDao.moveAccountBalance(
+              fromAccountId, toAccountId, new BigDecimal(request.getAmount()));
```

Comment:
new BigDecimal can throw if the 'amount' is not a valid numeric. The exception would be uncaught and the grpc client wouldn't get a nice error. This is pre-existing issue, we do that on createAccount as well. Consider creating an issue to better handle exceptions and ensure we send good messages to customers.

No need to fix with current PR as there are other similar issues.

Reply:
https://github.com/GoogleCloudPlatform/cloud-spanner-samples/issues/33